azurerm/resource_arm_managed_cluster.go

```diff
@@ -0,0 +1,556 @@
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmManagedCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedClusterCreate,
+		Read:   resourceArmManagedClusterRead,
+		Update: resourceArmManagedClusterCreate,
+		Delete: resourceArmManagedClusterDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"dns_prefix": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"kubernetes_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"linux_profile": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"admin_username": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ssh_key": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"key_data": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+				Set: resourceAzureRMManagedClusterLinuxProfilesHash,
+			},
+
+			"agent_pool_profile": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"count": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      1,
+							ValidateFunc: validation.IntBetween(1, 100),
+						},
+
+						"dns_prefix": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"fqdn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"vm_size": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"os_disk_size_gb": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  0,
+						},
+
+						"storage_profile": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(containerservice.StorageAccount),
+								string(containerservice.ManagedDisks),
+							}, true),
+						},
+
+						"vnet_subnet_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"os_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  containerservice.Linux,
```

Comment:
we need to cast this to a string here - also as Windows is also a possible option here can we add some validation with all the supported OS's? e.g.

```
ValidateFunc: validation.StringInSlice([]string{
  string(containerservice.Linux),
  string(containerservice.Windows),
}, true),
DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
```

Reply:
Thanks, added validation.